test/testuser.py

```diff
@@ -262,6 +261,11 @@ def get_active_devices(self):
                 active_devices.append(device)
         return active_devices
 
+    def query_devices(self, username=None, devices: List[TestDevice] = None, groups: List[str] = None):
```

Comment:
Nice use of type annotations :) 
We don't have anything checking them at present (we would need to start using mypy) so they're purely for documentation (which is fine and worthwhile)

Reply:
Should keep cam happy if he ever sees it